apps/nationaltoday/nationaltoday.star

```diff
@@ -0,0 +1,130 @@
+"""
+Applet: National Today
+Summary: Get National Today holidays
+Description: Displays today's holidays from National Today.
+Author: rs7q5 (RIS)
+"""
+
+#nationalToday.star
+#Created 20220130 RIS
+#Last Modified 20220201 RIS
+
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("cache.star", "cache")
+
+#load("schema.star","schema")
+load("math.star", "math")
+
+#load("time.star","time")
+load("re.star", "re")
+
+base_URL = "https://nationaltoday.com/what-is-today/"
+
+def main():
+    font = "tb-8"  #set font
+
+    #check for cached data
+    holidays_cached = cache.get("holiday_rate")
+    if holidays_cached != None:  #if any are None then all(title_cached)==False
+        print("Hit! Displaying cached data.")
+
+        holiday_txt = json.decode(holidays_cached)
+        holiday_fmt = format_text(holiday_txt, font)
+    else:
+        print("Miss! Calling NationalToday data.")  #error code checked within each function!!!!
+
+        #get the data
+        rep = http.get(base_URL)
+        if rep.status_code != 200:
+            holidays_list = re.findall('holiday-title">(.*?)<', rep.body())  #finds the holiday titles for the current day
+
+            #get the date text too
+            date = re.findall('meta name="description" content=(.*?)-', rep.body())[0]  #(.*?)/>',rep.body())
+            date = re.sub('meta name="description" content="| -', "", date)
+
+            #parse through and set up text for each holiday
+            holiday_txt = []
+            for i, holiday in enumerate(holidays_list):
+                holiday_txt.append(re.sub('holiday-title">|<', "", holiday))
+
+            if holiday_txt == []:
+                holiday_txt = ["No holidays today :("]
+
+            #shorten month name and add text to holidays
+            date_split = date.split(" ")
+            if date_split[0] in ("June", "July", "September"):
+                date_split[0] = date_split[0][:4]
+            else:
+                date_split[0] = date_split[0][:3]
+
+            holiday_txt.insert(0, " ".join(date_split))
+        else:
+            holiday_txt = ["Error", "Could not get holidays!!!!"]
+
+        #cache the data
+        cache.set("holiday_rate", json.encode(holiday_txt), ttl_seconds = 86400)  #grabs it once a day
```

Comment:
Please only cache for 30 minutes instead of an entire day. Otherwise we could, for example, fetch the page at 11 PM on Monday and keep serving Monday's response through 11 PM on Tuesday.
```suggestion
        cache.set("holiday_rate", json.encode(holiday_txt), ttl_seconds = 1800)  # cache for 30 min
```